core/src/services/dbfs/backend.rs

```diff
@@ -0,0 +1,519 @@
+
+        self.client.send_dbfs(req).await
+    }
+
+    async fn dbfs_get_properties(&self, path: &str) -> Result<Response<IncomingAsyncBody>> {
+        let p = build_rooted_abs_path(&self.root, path)
+            .trim_end_matches('/')
+            .to_string();
+
+        let url = format!(
+            "{}/api/2.0/dbfs/get-status?path={}",
+            &self.endpoint,
+            percent_encode_path(&p)
+        );
+
+        let mut req = Request::get(&url);
+
+        let auth_header_content = format!("Bearer {}", self.token);
+        req = req.header(header::AUTHORIZATION, auth_header_content);
+
+        let req = req
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
+
+        self.client.send(req).await
+    }
+
+    async fn dbfs_ensure_parent_path(&self, path: &str) -> Result<()> {
+        let resp = self.dbfs_get_properties(path).await?;
+
+        match resp.status() {
+            StatusCode::OK => return Ok(()),
+            StatusCode::NOT_FOUND => {
+                self.create_dir(path, OpCreateDir::default()).await?;
+            }
+            _ => return Err(parse_error(resp).await?),
+        }
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl Accessor for DbfsBackend {
+    type Reader = IncomingDbfsAsyncBody;
+    type BlockingReader = ();
+    type Writer = oio::OneShotWriter<DbfsWriter>;
+    type BlockingWriter = ();
+    type Pager = DbfsPager;
+    type BlockingPager = ();
+
+    fn info(&self) -> AccessorInfo {
+        let mut am = AccessorInfo::default();
+        am.set_scheme(Scheme::Dbfs)
+            .set_root(&self.root)
+            .set_native_capability(Capability {
+                stat: true,
+
+                read: true,
+                read_can_next: true,
+                read_with_range: true,
+
+                write: true,
+                create_dir: true,
+                delete: true,
+                rename: true,
+
+                list: true,
+                list_with_delimiter_slash: true,
+
+                ..Default::default()
+            });
+        am
+    }
+
+    async fn create_dir(&self, path: &str, _: OpCreateDir) -> Result<RpCreateDir> {
+        let req = self.dbfs_create_dir_request(path)?;
+
+        let resp = self.client.send(req).await?;
+
+        let status = resp.status();
+
+        match status {
+            StatusCode::CREATED | StatusCode::OK => {
+                resp.into_body().consume().await?;
+                Ok(RpCreateDir::default())
+            }
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn read(&self, path: &str, args: OpRead) -> Result<(RpRead, Self::Reader)> {
+        let resp = self.dbfs_read(path, args.range()).await?;
+
+        let status = resp.status();
+
+        match status {
+            StatusCode::OK => {
+                // NOTE: If range is not specified, we need to get content length from stat API.
+                if let Some(size) = args.range().size() {
```

Comment:
This might be wrong since we can read max to 1MiB. Our reader should handle those by:

- Got the whole length of file (return it as the content length here)
- Sending `/api/2.0/dbfs/read` call with 1MiB repeatedly until we reach the end or hits the range.

We can implement seek for this reader too.